lib/vmpooler/pool_manager.rb

```diff
@@ -459,69 +458,17 @@ def _check_snapshot_queue
       end
     end
 
-    def migration_limit(migration_limit)
-      # Returns migration_limit setting when enabled
-      return false if migration_limit == 0 || !migration_limit # rubocop:disable Style/NumericPredicate
-      migration_limit if migration_limit >= 1
-    end
-
     def migrate_vm(vm_name, pool_name, provider)
       Thread.new do
         begin
-          _migrate_vm(vm_name, pool_name, provider)
+          provider.migrate_vm(pool_name, vm_name, $redis)
         rescue => err
           $logger.log('s', "[x] [#{pool_name}] '#{vm_name}' migration failed with an error: #{err}")
-          remove_vmpooler_migration_vm(pool_name, vm_name)
+          provider.remove_vmpooler_migration_vm(pool_name, vm_name, $redis)
```

Comment:
This new method `remove_vmpooler_migration_vm` doesn't exist on the base and on inspection probably shouldn't be in a provider anyway.  It's just removing the VM from the task queue.

By doing my previous suggestion, it's not needed as the VM is removed from the queue prior to be actioned in the provider.